src/psyGen.py

```diff
@@ -783,7 +922,7 @@ def _find_position(self, children, position):
             position += 1
             if child == self:
                 return True, position
-            if isinstance(child, Loop):
+            if child.children:
```

Comment:
Yes, the routine was meant to return a unique id for each node but was coded to only recurse into loops with children (as they were the only examples at the time). But we now have directives as well and may add other things in the future. I changed the code so that it works generically. The new tests for dependence analysis test the corrected code as we use the global id when checking for forward or backward dependencies.